### PR TITLE
implement `/eth/v1/beacon/pool/attestations` POST

### PR DESCRIPTION
**Pull request by rolfyone**
- deprecated old `/validator/attestation` POST endpoint

 - switched validator client to use the new standard rest endpoint.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostAttestationData.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.POST,
+      summary = "Submit a signed attestation",
+      tags = {TAG_V1_VALIDATOR, TAG_VALIDATOR_REQUIRED},
+      requestBody = @OpenApiRequestBody(content = {@OpenApiContent(from = Attestation.class)}),
+      description =
+          "Submit a signed attestation to the beacon node to be validated and submitted if valid.\n\n"
+              + "This endpoint does not protected against slashing. Signing the attestation can result in a slashable offence.",
```
Comment (ajsutton) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostAttestationData.java]: The attestation is already signed in this case.  Maybe
```suggestion
              + "This endpoint does not protect against slashing.",
```

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/validator/PostAttestation.java [hunk 2]
```diff
@@ -54,12 +53,13 @@ public PostAttestation(final ValidatorDataProvider provider, final JsonProvider
...
-      summary = "Submit a signed attestation.",
+      summary = "Submit a signed attestation",
       tags = {TAG_VALIDATOR},
       requestBody = @OpenApiRequestBody(content = {@OpenApiContent(from = Attestation.class)}),
       description =
           "Submit a signed attestation to the beacon node to be validated and submitted if valid.\n\n"
-              + "This endpoint does not protected against slashing. Signing the attestation can result in a slashable offence.",
+              + "This endpoint does not protected against slashing. Signing the attestation can result in a slashable offence.\n\n"
+              + "Deprecated - use `/eth/v1/validator/attestation_data` instead.",
       responses = {
```
Comment (ajsutton) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/validator/PostAttestation.java]: Don't we normally add a `deprecated = true` option?

**Review by ajsutton — APPROVED** [review]
LGTM.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/validator/PostAttestation.java (hunk 2 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/validator/PostAttestation.java]: yep. done

**Pull request merged**